node/src/components/block_executor.rs

```diff
@@ -83,7 +84,11 @@ type BlockHeight = u64;
 /// The Block executor component.
 #[derive(DataSize, Debug, Default)]
 pub(crate) struct BlockExecutor {
-    genesis_state_root_hash: Option<Digest>,
+    /// Height of the highest known block
+    highest_block_height: Option<u64>,
+    /// Summary of the highest known block
+    highest_block_summary: Option<ExecutedBlockSummary>,
```

Comment:
This can be taken from the `parent_map`, right?

Reply:
Ah, it cannot b/c it's populated with information from the `highest_block_header` in the constructor.

Reply:
Not really. If we use the `parent_map`, we run into issues related to `commit_upgrade`. We will get a wrong pre-state-hash for the block right after an upgrade.